### PR TITLE
[systemtest] Fix failing tests

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lkral@redhat.com>

### Type of change

- Bugfix

### Description

This PR will fix these tests:
```
testRemoveUserAndTopicOperatorsFromEntityOperator - test waited for pod deletion with bad name
testCustomCertLoadBalancerAndTlsRollingUpdate - asserts with bad message counts
testJvmAndResources - update test - before changes were all `-D` and -`X..` options in env`STRIMZI_JAVA_SYSTEM_PROPERTIES`, now it's moved like this: `-D` opt in `STRIMZI_JAVA_SYSTEM_PROPERTIES` and `-X..` options in `STRIMZI_JAVA_OPTS`
```

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ListenersST#testCustomCertLoadBalancerAndTlsRollingUpdate,KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator+testJvmAndResources

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java [hunk 1]
```diff
@@ -657,12 +657,15 @@ void testCustomCertLoadBalancerAndTlsRollingUpdate() throws Exception {
...
 
         internalKafkaClient.setConsumerGroup("consumer-group-certs-71");
-        internalKafkaClient.setMessageCount(MESSAGE_COUNT * 3);
+        internalKafkaClient.setMessageCount(MESSAGE_COUNT);
 
         sent = internalKafkaClient.sendMessagesTls();
-        assertThat(sent, is(3 * MESSAGE_COUNT));
+        assertThat(sent, is(MESSAGE_COUNT));
+
+        internalKafkaClient.setMessageCount(MESSAGE_COUNT * 5);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java:665]: Why 5 times?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 2]
```diff
@@ -1009,34 +1018,31 @@ void testRemoveUserAndTopicOperatorsFromEntityOperator() {
...
         //Waiting when EO pod will be deleted
-        DeploymentUtils.waitForDeploymentDeletion(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME));
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME));
-        PodUtils.waitForPodDeletion(eoPodName);
+        DeploymentUtils.waitForDeploymentDeletion(eoDeploymentName);
+        ReplicaSetUtils.waitForReplicaSetDeletion(eoDeploymentName);
 
+        if (kubeClient().listPodsByPrefixInName(eoDeploymentName).size() != 0) {
+            PodUtils.waitForPodDeletion(kubeClient().listPodsByPrefixInName(eoDeploymentName).get(0).getMetadata().getName());
+        }
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:1034]: Wouldn't it be better to check this list size in `waitForPodDeletion()` ?

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java:665]: I need to get some info from Maros about clients -> but if you set different CONSUMER_GROUP you'll receive all messages until that moment -> not just messages you set in setMessagesCount. So maybe it's problem in VerifiableClient. But in the original test (before refactor of the clients) was it this way.

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java (hunk 2 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:1034]: Now it's redundant here

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java:665]: Can you open the issue for it? We should understand what's happening here.

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/ListenersST.java:665]: Yeah sure -> how I said - maybe it's fault in understanding from my side, we will see.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ListenersST#testCustomCertLoadBalancerAndTlsRollingUpdate,KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator+testJvmAndResources

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 3]
```diff
@@ -845,12 +845,21 @@ void testJvmAndResources() {
...
-                if (container.getName().equals("user-operator"))
-                    assertThat(value, is("-Xms512M -Xmx1G -Djavax.net.debug=verbose"));
+
+                String javaSystemProp = container.getEnv().stream().filter(envVar ->
+                    envVar.getName().equals("STRIMZI_JAVA_SYSTEM_PROPERTIES")).findFirst().get().getValue();
+                String javaOpts = container.getEnv().stream().filter(envVar ->
+                    envVar.getName().equals("STRIMZI_JAVA_OPTS")).findFirst().get().getValue();
+
+                if (container.getName().equals("topic-operator")) {
+                    assertThat(javaSystemProp, is("-Djavax.net.debug=verbose"));
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java:855]: this assert can be written just one time outside of the `if`(s), it's common to both.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ListenersST#testCustomCertLoadBalancerAndTlsRollingUpdate,KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator+testJvmAndResources

**Comment by im-konge** [comment]
@scholzj after tests will be successful, PR is ready for merge :)

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ListenersST#testCustomCertLoadBalancerAndTlsRollingUpdate,KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator+testJvmAndResources

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator

**Comment by im-konge** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaST#testRemoveUserAndTopicOperatorsFromEntityOperator

**Pull request merged**